src/dmd/statementsem.d

```diff
@@ -3084,6 +3084,16 @@ else
             if (rs.exp.op == TOK.type)
                 rs.exp = resolveAliasThis(sc, rs.exp);
 
+            if (rs.exp.op == TOK.scope_)
```

Comment:
This being checked for only in return statements seems awfully specific. What if `modulename` simply appears in an expression? Or other statement type?